classes/class-wc-connect-service-settings-store.php

```diff
@@ -110,6 +110,45 @@ public function can_user_manage_payment_methods() {
 				( is_a( $master_user, 'WP_User' ) && $current_user->ID === $master_user->ID );
 		}
 
+		public function get_end_shipper_email() {
+			$master_user = WC_Connect_Jetpack::get_master_user();
+
+			if ( $master_user instanceof WP_User ) {
+
+				$connected_data = WC_Connect_Jetpack::get_connected_user_data( $master_user->ID );
+
+				if ( is_array( $connected_data ) && ! empty( $connected_data['email'] ) ) {
+					return $connected_data['email'];
+				}
+			}
+
+			return '';
+		}
+
+		/**
+		 * Gets the store's location settings.
+		 *
+		 * @return array
+		 */
+		public function get_end_shipper_address() {
+			$store_settings = array(
+				'company'   => html_entity_decode( get_bloginfo( 'name' ), ENT_QUOTES ), // HTML entities may be saved in the option.
+				'address_1' => WC()->countries->get_base_address(),
+				'address_2' => WC()->countries->get_base_address_2(),
+				'city'      => WC()->countries->get_base_city(),
+				'state'     => WC()->countries->get_base_state(),
+				'country'   => WC()->countries->get_base_country(),
+				'postcode'  => WC()->countries->get_base_postcode(),
+				'email'     => $this->get_end_shipper_email(),
+
+				// Phone and name data will be added later from origin address.
+				'phone'     => '',
+				'name'      => '',
+			);
+
+			return apply_filters( 'end_shipper_store_settings', $store_settings );
```

Comment:
Just to avoid running into collisions with any potential other extensions doing this.

```suggestion
			return apply_filters( 'wcship_end_shipper_store_settings', $store_settings );
```